### PR TITLE
Fix infinite recursion in exception instrumentation

### PR DESCRIPTION
**Pull request by mar-kolya**
The idea is that class loading in of itself is reentrant so we just
need to avoid infinite synchronous loading in the same thread

### CODE REVIEW

File: dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java [hunk 1]
```diff
@@ -8,15 +8,30 @@
  */
 public final class ExceptionProfiling {
 
-  private static final ExceptionProfiling INSTANCE = new ExceptionProfiling(Config.get());
+  private static ExceptionProfiling instance;
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java:11]: Sorry to bikeshed, but why the case change?

Comment (mar-kolya) [reply dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java:11]: It's no longer `final` constant

Comment (mar-kolya) [reply dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java:11]: but I can change it back :)

File: dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java [hunk 2]
```diff
@@ -8,15 +8,30 @@
...
+  }
 
   /**
    * Get a pre-configured shared instance.
    *
    * @return the shared instance
    */
   public static ExceptionProfiling getInstance() {
-    return ExceptionProfiling.INSTANCE;
+    return ExceptionProfiling.instance;
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java:34]: Are callers able to handle null until this is initialized?

Comment (mar-kolya) [reply dd-java-agent/instrumentation/exception-profiling/src/main/java11/com/datadog/profiling/exceptions/ExceptionProfiling.java:34]: Callers are able... but this doesn't solve original problem :(

**Comment by tylerbenson** [comment]
Can this be closed?

**Comment by mar-kolya** [comment]
Yeah

**Pull request closed without merging**